### PR TITLE
Doc for Custom Elasticsearch Index

### PR DESCRIPTION
**Pull request by DaRaFF**
Issue: https://github.com/livingdocsIO/livingdocs-planning/issues/3610
PR: https://github.com/livingdocsIO/livingdocs-server/pull/3185

### CODE REVIEW

File: reference-docs/server-configuration/custom-index.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+- index metadata in another format
+- index resolved document references
+- optimised search queries
+- index other data than publications (e.g. data from your own feature)
+
+## Setup a Custom Index
+
+An integration is based on the server config and the initialisation file which contains the transformation, thats it!
+
+First, one needs to setup the configuration. You can see an example below.
```
Comment (meinradandermatt) [inline reference-docs/server-configuration/custom-index.md:28]: I would suggest something like `First, the elasticIndex server config needs to be added.`

File: reference-docs/server-configuration/custom-index.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+      handle: 'my-custom-publication',
+
+      // file to define the mapping and the transformation of the documents
+      indexInitializationFile: require.resolve('../../app/search/my-custom-publication/init.js'),
+
+
+      // optional
+      // --------
+
+      // The context is passed to the 'processBatch' and 'createBatches' function
```
Comment (meinradandermatt) [inline reference-docs/server-configuration/custom-index.md:66]: Is the context also used from the default `createBatches` implementation?

File: reference-docs/server-configuration/custom-index.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+  /**
+   * Process a batch of documents
+   *   1) Load batch of documents/publications
+   *   2) Map documents/publications to Elasticsearch format
+   *   3) Index documents/publications into Elasticsearch
+   *
+   * Both use cases
+   *   - background indexing via CLI
+   *   - live indexing (e.g. press the publish button in the editor)
```
Comment (meinradandermatt) [inline reference-docs/server-configuration/custom-index.md:152]: Where is the configuration to set this for the publication?

File: reference-docs/server-configuration/custom-index.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+   * @param {?string}       params.context.contentType
+   * @param {?string}       params.context.documentType
+   * @param {?number}       params.context.projectId
+   * @param {Object}        params.context.updatedAt
+   * @param {number|Date}   params.context.updatedAt.from time in ms since 1970
+   * @param {?}             params.context.myCustomValue - passed via context object of index config
+   */
+  async function processBatch ({context, range, ids}) {
+    const documentVersions = await publicationApi.getLatestPublicationsV2({...context, ...range, ids})
+    const updatedDocumentVersions = await Promise.all(documentVersions.map((d) => metadataApi.updateOnRender(d)))
```
Comment (meinradandermatt) [inline reference-docs/server-configuration/custom-index.md:171]: What happens on en error?

File: reference-docs/server-configuration/custom-index.md [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+  }
+}
+```
+
+
+## Server API
+
+### Search
+
+Based on the customIndex configs, Elasticsearch indexes are updated via live indexing (automatically after the publish event) or via background indexing (reindexing via CLI). It's possible to search in your custom index with that API below:
```
Comment (meinradandermatt) [inline reference-docs/server-configuration/custom-index.md:236]: Do you have an example for what this can be used?

**Review by meinradandermatt — APPROVED** [review]
Looks good, I have some small improvements or questions.

File: reference-docs/server-configuration/custom-index.md [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+      // When disabled, the index will be ignored for all operations
+      enabled: true, // default: true
+
+      // Overwrite the alias pointing to your elastic index
+      // The default alias is the 'handle' (in this example - 'my-custom-publication')
+      alias: 'an-alias'
+
+      // enable/disable the Livingdocs publication index (used in the public API for search requests)
+      // see: [Publication Index](../server-configuration/publication-index.md)
+      documentPublicationIndexEnabled: true, // default: true
```
Comment (gabrielhase) [inline reference-docs/server-configuration/custom-index.md:86]: this option was in here twice

File: reference-docs/server-configuration/custom-index.md (hunk 2 above)
Comment (DaRaFF) [reply reference-docs/server-configuration/custom-index.md:66]: yes

File: reference-docs/server-configuration/custom-index.md (hunk 4 above)
Comment (DaRaFF) [reply reference-docs/server-configuration/custom-index.md:171]: The error will be catched and logged in the processing pipeline. The job batch fails.

File: reference-docs/server-configuration/custom-index.md (hunk 3 above)
Comment (DaRaFF) [reply reference-docs/server-configuration/custom-index.md:152]: I'm not sure if I understand the question. As soon as you register a custom index, the live indexing integration automatically calls the custom index.

File: reference-docs/server-configuration/custom-index.md (hunk 5 above)
Comment (DaRaFF) [reply reference-docs/server-configuration/custom-index.md:236]: When you have data in your custom index, maybe you also have your custom feature with a public API and you want to provide a search over the custom index. 
As a concrete example: NZZ used this search function to kick off a new batch job when a list hook was called.

**Pull request merged**